app/models/manageiq/providers/vmware/infra_manager/vm_or_template_shared/operations.rb

```diff
@@ -10,10 +10,10 @@ def raw_set_custom_field(attribute, value)
   end
 
   def raw_clone(name, folder, pool = nil, host = nil, datastore = nil, powerOn = false, template_flag = false, transform = nil, config = nil, customization = nil, disk = nil)
-    folder_mor    = folder.ems_ref_obj    if folder.respond_to?(:ems_ref_obj)
+    folder_mor    = VimString.new(folder.ems_ref, folder.ems_ref_type, :ManagedObjectReference) if folder
     pool_mor      = pool.ems_ref_obj      if pool.respond_to?(:ems_ref_obj)
     host_mor      = host.ems_ref_obj      if host.respond_to?(:ems_ref_obj)
-    datastore_mor = datastore.ems_ref_obj if datastore.respond_to?(:ems_ref_obj)
+    datastore_mor = VimString.new(datastore.ems_ref, datastore.ems_ref_type, :ManagedObjectReference) if datastore
```

Comment:
Ditto on these 2 as well (and probably any others)

Reply:
So Storages we can't subclass by provider because they currently are able to be cross-ems and there's nothing that enforces that they only belong to the same ems type.